location_obfuscation/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use geo_types::geometry::{Point};
+//use rayon::prelude::*;
+use rstar::primitives::GeomWithData;
+use rstar::{PointDistance, RTreeObject};
+use rusty_roads::{Id, NearestNeighbor, Queryable, RoadKey};
+use std::collections::VecDeque;
+use thiserror::Error;
+
+const WINDOW_SIZE: usize = 5; // Must be an odd number.
+
+#[derive(Error, Debug)]
+pub enum LocationObfuscationError {
+    #[error("Cannot do obfuscation when no points are provided")]
+    NoPointsProvided,
+}
+
+pub fn obfuscate_points<T, U, V>(points: T, roads: V) -> Result<Vec<Point>, LocationObfuscationError>
+where
+    T: Iterator<Item = U> + Clone,
+    U: PointDistance + RTreeObject + Clone,
+    V: NearestNeighbor<U, GeomWithData<U, Id>> + Queryable<RoadKey>,
+{
+    let mut ids: VecDeque<Id> = points.clone()
+        .filter_map(|x| roads.nearest_neighbor(x))
+        .map(|x| x.data)
+        .collect();
+
+    let first = if let Some(element) = ids.front() {
+        *element
+    } else {
+        return Err(LocationObfuscationError::NoPointsProvided);
+    };
```

Comment:
```suggestion
    let first = ids.front().ok_or(LocationObfuscationError::NoPointsProvided)?;
```
er selv større fan af denne stil, men kan forestille mig at du er uenig :(